### PR TITLE
build: update to kvdb v1.4.3

### PR DESCRIPTION
**Pull request by Roasbeef**
We needed a tag after updating the module with retry for sqlite/postgres.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -39,7 +39,7 @@ require (
 	github.com/lightningnetwork/lnd/cert v1.2.1
 	github.com/lightningnetwork/lnd/clock v1.1.0
 	github.com/lightningnetwork/lnd/healthcheck v1.2.2
-	github.com/lightningnetwork/lnd/kvdb v1.4.2
+	github.com/lightningnetwork/lnd/kvdb v1.4.3
```
Comment (guggero) [inline go.mod:42]: If we're bumping the `kvdb` version in this PR already, can we please also bump all other submodules? We made changes to all of them, so I pushed the following tags:
```
cert/v1.2.2
clock/v1.1.1
healthcheck/v1.2.3
queue/v1.1.1
ticker/v1.1.1
tlv/v1.1.1
tor/v1.1.2
```

File: kvdb/sqlbase/sqlerrors_no_postgres.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+//go:build !kvdb_postgres
+// +build !kvdb_postgres
```
Comment (guggero) [inline kvdb/sqlbase/sqlerrors_no_postgres.go:2]: I think we should start to remove the old style build tags everywhere, or at least not use them in new code anymore.

File: kvdb/sqlbase/sqlerrors_sqlite.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
+//go:build kvdb_sqlite && !(windows && (arm || 386)) && !(linux && (ppc64 || mips || mipsle || mips64))
```
Comment (guggero) [inline kvdb/sqlbase/sqlerrors_sqlite.go:1]: :+1:

**Review by guggero — APPROVED** [review]
LGTM :tada: 

Would be nice to bump the other gomodules in this PR as well to make sure we don't need yet another PR with changes.

File: go.mod (hunk 1 above)
Comment (yyforyongyu) [reply go.mod:42]: we'd also need to bump `btcutil` and `btcd/btcutil`

**Review by yyforyongyu — COMMENTED** [review]
Need to remove `replace github.com/lightningnetwork/lnd/kvdb => ./kvdb` otherwise good to go🙏

**Comment by Roasbeef** [comment]
> Need to remove replace github.com/lightningnetwork/lnd/kvdb => ./kvdb otherwise good to go🙏

Can't remove this, since I had to modify the build tags in `kvdb` package. Or I can do it in another PR, tag the version, then update this. 

Otherwise plan was: merge this, tag, then new PR to do rc3 and update the module.

**Review by guggero — COMMENTED** [review]
Needs one more tweak to get the itests passing, then we're good to go:

```diff
diff --git a/itest/lnd_channel_funding_utxo_selection_test.go b/itest/lnd_channel_funding_utxo_selection_test.go
index f52af7a41..27c831351 100644
--- a/itest/lnd_channel_funding_utxo_selection_test.go
+++ b/itest/lnd_channel_funding_utxo_selection_test.go
@@ -91,7 +91,7 @@ func testChannelUtxoSelection(ht *lntest.HarnessTest) {
                        feeRate:            15,
                        expectedErrStr: "output amount(0.00000174 BTC) after " +
                                "subtracting fees(0.00001826 BTC) below dust " +
-                               "limit(0.0000033 BTC)",
+                               "limit(0.00000330 BTC)",
                },
                // Selected coins don't cover the minimum channel size.
                {
@@ -102,7 +102,7 @@ func testChannelUtxoSelection(ht *lntest.HarnessTest) {
                        feeRate:            1,
                        chanOpenShouldFail: true,
                        expectedErrStr: "available funds(0.00017877 BTC) " +
-                               "below the minimum amount(0.0002 BTC)",
+                               "below the minimum amount(0.00020000 BTC)",
                },
                // The local amount exceeds the value of the selected coins.
                {
@@ -114,7 +114,7 @@ func testChannelUtxoSelection(ht *lntest.HarnessTest) {
                        chanOpenShouldFail: true,
                        expectedErrStr: "not enough witness outputs to " +
                                "create funding transaction, need 0.00210337 " +
-                               "BTC only have 0.001 BTC available",
+                               "BTC only have 0.00100000 BTC available",
                },
                // We are spending two selected coins partially out of three
                // available in the wallet and expect a change output and the

```

**Review by yyforyongyu — COMMENTED** [review]
Pending @guggero 's comments, then this is good to go🏖

> Otherwise plan was: merge this, tag, then new PR to do rc3 and update the module

This works. Just need to follow up closely as otherwise the linter will fail for other rebased PRs.

**Comment by yyforyongyu** [comment]
Unit tests for both the SQL backends failed, are those flakes?

**Pull request merged**